examples/llama/2d_llama.py

```diff
@@ -0,0 +1,105 @@
+# $ torchrun --nproc-per-node 8 2d_llama.py
+import os
+import torch
+from transformers import AutoModelForCausalLM, AutoTokenizer
+from pippy import Pipe, PipeSplitWrapper, annotate_split_points, PipelineStage
+from torch.distributed._tensor import init_device_mesh
+
+
+# Utility
+def modify_view(
```

Comment:
hmmmm this is a very dangerous operation, we shouldn't compose PP + TP like this IMO. In particular this would modify any view operations in the traced graph and thereby making the graph be super unsafe to the user, also this is not scalable to another models especially non-llama model might have view operations in non-attention layers, this would either trigger the assertion failure or wrongly modify the view ops.

Reply:
Thanks much @wanchaol for the comments. I agree with your thoughts. `modify_view` is more of a workaround than a general solution. For a better one, I wonder if the following direction is worth consideration. 
Today, for TP to work well with attention, view ops of user code needs to be modified. This is true even for eager mode. I wonder if it would be possible for the output of ColwiseParalell to play well with unchanged view ops. This may require: the output of ColwiseParalell to stay in some kind of DTensor form (rather than a regular tensor), and for this DTensor form to have special rule with the view ops (i.e. recognizing that some dimension of the view ops needs to be adjusted based on how the source operation is distributed).
Any thoughts here?
Cc @fduwjj

Reply:
Removed modify_view() now